tests/base/boring_model.py

```diff
@@ -75,7 +82,14 @@ def training_step(...):
 
         """
         super().__init__()
-        self.layer = torch.nn.Linear(32, 2)
+        self.layer = torch.nn.Linear(in_features, out_features)
+        self.batch_size = batch_size
+        self.in_features = in_features
+        self.learning_rate = learning_rate
+        self.optimizer_name = optimizer_name
+        self.out_features = out_features
+        self.save_hyperparameters('batch_size', 'in_features', 'out_features',
+                                  'optimizer_name', 'learning_rate')
```

Comment:
```suggestion
        self.save_hyperparameters()
```
you can just use `self.save_hyperparameters()` and access them using `self.hparams.argument`.

Reply:
Hi @rohitgr7, I have a question regarding `save_hyperparameters()`. In particular, if I call it in __init__ method of `BoringModel`, than subclass model `RuntimeParamChangeModelAssign` in `test_hparams.py` fail to save their hypermaremeters (specifically if args are assigned to `self.hparams` as in line 601 in 'tests/models/test_hparams.py')

Reply:
assigning `hparams` like `self.hparams = params` is not the recommended way and was deprecated due to this reason only AFAIR. use `self.save_hyperparameters()` there too.

Reply:
```python
class RuntimeParamChangeModelAssign(BoringModel):
    def __init__(self, **kwargs):
        super().__init__()
        self.save_hyperparameters()
```

Reply:
Just as a side note for completeness sake:
It is totally fine to assign `self.something = something`. The only exception is self.hparams, that was an attribute with special behaviour and this is now deprecated. save_hyperparameters() will bundle all args into a namespace/dict accessible through self.hparams. It is up to the user to decide if they want to read from it or not.

both the current changes and the ones proposed by @rohitgr7 look good to me :)

Reply:
Thank you both for the explanations! :) :top: